File: Java/Warmup-2/stringSplosion.java

public class warmup2 {

	// given a non-empty string like "Code" return a string like "CCoCodCode"
	public String stringSplosion(String str) {
	
		String s = "";
		
		for (int i = 0; i < str.length(); i++) {
			s = s + str.substring(0, i+1);
		}
		
		return s;

	}
	
	public static void main(String[] args) {
		
		warmup2 o = new warmup2();
		
		System.out.println(o.stringSplosion("Code"));
		System.out.println(o.stringSplosion("abc"));
		System.out.println(o.stringSplosion("ab"));

	}

}

File: JavaScript/Warmup-2/stringSplosion.js

// given a non-empty string like "Code" return a string like "CCoCodCode"
const stringSplosion = (str) => {

	let newStr = '';

	for (let i = 0; i < str.length; i++) {
		newStr = newStr + str.substring(0, i+1);
	}

	return newStr;

};

console.log(stringSplosion('Code'));
console.log(stringSplosion('abc'));
console.log(stringSplosion('ab'));
